crates/ruff_python_semantic/src/helpers.rs

```diff
@@ -0,0 +1,37 @@
+use crate::SemanticModel;
+use ruff_python_ast as ast;
+use ruff_python_ast::Expr;
+use ruff_text_size::Ranged;
+
+/// Returns `true` if the given expression is either an unused value or a tuple of unused values.
+pub fn is_unused(expr: &Expr, semantic: &SemanticModel) -> bool {
```

Comment:
I wonder if this could be made a method on the `SemanticModel`:
```rust
impl SemanticModel {
	// ...

	pub fn is_expr_unused(&self, expr: &Expr) -> bool {
		// ...
	}	
}

checker.semantic().is_expr_unused(expr)
```

But then it would mean that we would need to visit all relevant expressions in this context which I think is out of scope for this PR.